MetaMask/metamask-extension
# fix: AmonHenV2 connection flow incremental permitted chain approval and account address case comparison

## **Description**

Previously the AmonHenV2 incremental permitted chain approval view was not correctly showing the chain being permitted. This PR fixes this correctly dereferencing the caveat value. 

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27518?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. `CHAIN_PERMISSIONS=1 yarn start` 
2. On a dapp that is connected, open the console
3. 
```
await window.ethereum.request({
 "method": "wallet_switchEthereumChain",
 "params": [
  {
    chainId: "0x1" // use a chain that is not already permitted for the dapp
  }
],
});
```
4. Expect to see the network being switched to in the component

## **Screenshots/Recordings**



### **Before**



### **After**

![Screenshot 2024-09-30 at 3 13 33 PM](https://github.com/user-attachments/assets/cc050a5c-dc1c-4bed-8e35-1799f4ef131d)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
